src/main/java/com/esri/samples/na/find_service_areas_for_multiple_facilities/FindServiceAreasForMultipleFacilitiesSample.java

```diff
@@ -0,0 +1,224 @@
+              findServiceAreasButton.setDisable(false);
+              mapView.removeDrawStatusChangedListener(this);
+            }
+          }
+        });
+
+        // resolve 'find service areas' button click
+        findServiceAreasButton.setOnAction(event -> {
+
+          // disable the button
+          findServiceAreasButton.setDisable(true);
+
+          // show the progress indicator
+          progressIndicator.setVisible(true);
+
+          // create a service area task from URL
+          ServiceAreaTask serviceAreaTask = new ServiceAreaTask("https://sampleserver6.arcgisonline.com/arcgis/rest/services/NetworkAnalysis/SanDiego/NAServer/ServiceArea");
+          serviceAreaTask.loadAsync();
+
+          // create default service area task parameters
+          ListenableFuture<ServiceAreaParameters> serviceAreaTaskParametersFuture = serviceAreaTask.createDefaultParametersAsync();
+          serviceAreaTaskParametersFuture.addDoneListener(() -> {
+            try {
+              ServiceAreaParameters serviceAreaParameters = serviceAreaTaskParametersFuture.get();
+              // set the task parameters to have the task return polygons
+              serviceAreaParameters.setPolygonDetail(ServiceAreaPolygonDetail.HIGH);
+              serviceAreaParameters.setReturnPolygons(true);
+              // add a service area of 3 minutes in addition to the default service area of 5 minutes
+              serviceAreaParameters.getDefaultImpedanceCutoffs().add(3.0);
+
+              // create query parameters used to select all facilities from the feature table
+              QueryParameters queryParameters = new QueryParameters();
+              queryParameters.setWhereClause("1=1");
+
+              // add all facilities to the service area parameters
+              serviceAreaParameters.setFacilities(facilitiesTable, queryParameters);
+
+              // find the service areas around the facilities using the parameters
+              ListenableFuture<ServiceAreaResult> serviceAreaResultFuture = serviceAreaTask.solveServiceAreaAsync(serviceAreaParameters);
+              serviceAreaResultFuture.addDoneListener(() -> {
+                try {
+                  // get the task results
+                  ServiceAreaResult serviceAreaResult = serviceAreaResultFuture.get();
+
+                  // display all the service areas that were found to the map view
+                  List<Graphic> serviceAreaGraphics = serviceAreasGraphicsOverlay.getGraphics();
+
+                  // iterate through all the facilities to get the service area polygons
+                  for (int facilityIndex = 0; facilityIndex < serviceAreaResult.getFacilities().size(); facilityIndex++) {
+                    List<ServiceAreaPolygon> serviceAreaPolygonList = serviceAreaResult.getResultPolygons(facilityIndex);
+                    // we may have more than one resulting service area, so create a graphics from each available polygon
```

Comment:
generally avoid using first person wording in comments. If I've understood the meaning of this comment correctly, can replace with "create a graphic for each available polygon, there may be more than one for each service area"